status/assign_roles.py

```diff
@@ -0,0 +1,89 @@
+import tornado.web
+import json
+import time
+import copy
+import base64
+import requests
+
+from dateutil import parser
+from status.util import SafeHandler, UnsafeHandler
+from status.authorization import UnAuthorizedHandler
+
+
+class AssignRolesHandler(SafeHandler):
+    """ Serves a page with users and roles listed, with the option to create new users
+    URL: /assign_roles
+    """
+    def get(self):
+        t = self.application.loader.load("assign_roles.html")
+        self.write(t.generate(gs_globals=self.application.gs_globals, user=self.get_current_user(),
+        roles=self.application.genstat_defaults['roles']))
+
+
+class AssignRolesUsersHandler(SafeHandler):
+    """Serves the data for populating user roles table and also methods to modify user roles
+    URL: /api/v1/assign_roles/users
+    """
+    def get(self):
+        self.set_header("Content-type", "application/json")
+        view_result={}
+        for row in self.application.gs_users_db.view('authorized/roles'):
+            view_result[row.key]=row.value
+        self.write(view_result)
+
+    def post(self):
+        data = json.loads(self.request.body)
+        userToChange = data['username']
+        headers = {"Accept": "application/json",
+                   "Authorization": "Basic " + "{}:{}".format(base64.b64encode(bytes(self.application.settings.get("username", None), 'ascii')),
+                   base64.b64encode(bytes(self.application.settings.get("password", None), 'ascii')))}
+
+        idtoChange=''
+        for row in self.application.gs_users_db.view('authorized/users'):
+            if row.get('key') == userToChange:
+                idtoChange = row.get('value')
+        action = self.get_argument('action')
+        if self.get_current_user().role == 'admin':
+            if action == 'create':
+                if idtoChange:
+                    self.set_status(409)
+                    self.write('User already exists!')
+                else:
+                    try:
+                        self.application.gs_users_db.save(data)
+                    except Exception as e:
+                        self.set_status(400)
+                        self.write(e.message)
```

Comment:
Are you sure the function stops executing here after this exception? I would add a `return` statement to be sure.

Reply:
Changed to self.finish()